From 4b7c7f5eec14ac5c2fdc90b23b64cf32cd8df95f Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 12 May 2022 10:46:21 +0200
Subject: [PATCH 187/954] feat(stm32mp2): add define for device unique id
 access

Add the defines UID_OTP and UID_WORD_NB to use the 3 OTPs
which contains the Unique  device ID register (96 bits) (UID).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e2e14f78dac9d0be7d18763deff8b38200581e17 feat(stm32mp2): add define for device unique id access]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ib3f11a590186b929ac73c000f569578e6196f2d3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/255794
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 84d843b23f..7cb6239596 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -264,6 +264,7 @@ enum ddr_type {
 #define NAND_OTP				"otp16"
 #define NAND2_OTP				"otp20"
 #define BOARD_ID_OTP				"board_id"
+#define UID_OTP					"uid_otp"
 
 /* OTP mask */
 /* PACKAGE */
@@ -325,6 +326,8 @@ enum ddr_type {
 /* MONOTONIC OTP */
 #define MAX_MONOTONIC_VALUE			32
 
+/* UID OTP */
+#define UID_WORD_NB				U(3)
 /*******************************************************************************
  * STM32MP2 TAMP
  ******************************************************************************/
-- 
2.53.0

